core/test/lib/libledger-test/FakeUrlConnection.cpp

```diff
@@ -0,0 +1,39 @@
+#include "FakeUrlConnection.hpp"
+#include "api/HttpReadBodyResult.hpp"
+#include "utils/optional.hpp"
+#include "api/Error.hpp"
+
+namespace ledger {
+namespace core {
+namespace test {
+    std::shared_ptr<FakeUrlConnection> FakeUrlConnection::fromString(const std::string& responseData) {
+        UrlConnectionData data;
+        data.body = responseData;
+        data.statusCode = 200;
+        data.statusText = "OK";
+        return std::make_shared<FakeUrlConnection>(data);
+    }
+
+    FakeUrlConnection::FakeUrlConnection(const UrlConnectionData& data)
+        : _data(data) {};
+
+    int32_t FakeUrlConnection::getStatusCode() {
```

Comment:
Do you really need all that code for tests? If so, why not make that function `const`?

Reply:
This is implementation of the interface `api::HttpUrlConnection`, so have not a lot of choises